### PR TITLE
add libsystemd/246.6

### PR DESCRIPTION
**Pull request by bobrofon**
Specify library name and version:  **libsystemd/246.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Croydon** [comment]
I'm not all against merging this as there might be good use cases for it

However,  I think we should not depend on it from other CCI recipes and should rather use there a `systemd/system` package

**Comment by bobrofon** [comment]
> However, I think we should not depend on it from other CCI recipes and should rather use there a `systemd/system` package

My intention to use libsystemd is a linking with it's dbus (sd-bus) implementation. This is unnecessary to require `systemd/system` in this case, and i would like not to have it in recipes if it doesn't really needed.
Well, using libsystemd package just for sd-bus is not a good solution. But I don't know how to improve it :disappointed:.

File: recipes/libsystemd/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "0.29.0":
+    url: "https://github.com/systemd/systemd-stable/archive/v246.6.tar.gz"
```
Comment (prince-chrismc) [inline recipes/libsystemd/all/conandata.yml:3]: I am curious... why these are different?

Comment (bobrofon) [reply recipes/libsystemd/all/conandata.yml]: Hm. Looks like i misunderstood about libsystemd versioning.
0.29.0 is a libsystemd.so version.
246.6 is a systemd source package version.

libsystemd.pc file contains systemd version. So the version should be 246.6 in both places.

File: recipes/libsystemd/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibsystemdConan(ConanFile):
+    name = "libsystemd"
+    license = "LGPL-2.1-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.freedesktop.org/wiki/Software/systemd/"
+    description = "systemd library"
```
Comment (uilianries) [inline recipes/libsystemd/all/conanfile.py:14]:
```suggestion
    description = "System and Service Manager"
```

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — DISMISSED** [review]

**Comment by Croydon** [comment]
> My intention to use libsystemd is a linking with it's dbus (sd-bus) implementation. This is unnecessary to require systemd/system in this case, and i would like not to have it in recipes if it doesn't really needed.
Well, using libsystemd package just for sd-bus is not a good solution. But I don't know how to improve it 😞.

My concern is that it might conflict with a the actual systemd installed on the target computer

File: recipes/libsystemd/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        src_folder = "systemd-stable-{}".format(self.version)
+        os.rename(src_folder, self._source_subfolder)
+
+    @property
+    @functools.lru_cache()
```
Comment (madebr) [inline recipes/libsystemd/all/conanfile.py:75]: `functools.lru_cache` is not available in python 2.7.
Yes, recipes of cci are supposed to build in 2.7.
```suggestion
```

File: recipes/libsystemd/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
+import functools
```
Comment (madebr) [inline recipes/libsystemd/all/conanfile.py:1]:
```suggestion
```

File: recipes/libsystemd/all/conanfile.py (hunk 3 above)
Comment (bobrofon) [reply recipes/libsystemd/all/conanfile.py]: > Yes, recipes of cci are supposed to build in 2.7.

:fearful: How hard is to be alive in 2020.

**Review by uilianries — APPROVED** [review]

**Comment by Croydon** [comment]
> My intention to use libsystemd is a linking with it's dbus (sd-bus) implementation. This is unnecessary to require systemd/system in this case, and i would like not to have it in recipes if it doesn't really needed.
Well, using libsystemd package just for sd-bus is not a good solution. But I don't know how to improve it 😞.

I would not worry about this too much. With components you don't have to link against everything, and if a system uses systemd the dependencies will be already installed anyway

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**